diesel/src/query_builder/bind_collector.rs

```diff
@@ -46,6 +46,23 @@ pub trait BindCollector<'a, DB: TypeMetadata>: Sized {
     }
 }
 
+/// Bind data from a BindCollector, that can be moved across threads
+pub trait MoveableBindData: Send + 'static {}
+
+/// A movable version of the bind collector which allows it to be extracted, moved and refilled.
+///
+/// This is mostly useful in async context where bind data needs to be moved across threads.
+pub trait MoveableBindCollector<DB: TypeMetadata> {
```

Comment:
I would like to gate that behind the `i-implement-a-third-party-backend-and-opt-into-breaking-changes` feature, as it's not useful for anyone outside of diesel-async/other crates doing really low level stuff.